packages/components/src/CollapsiblePanel/CollapsiblePanel.component.js

```diff
@@ -84,107 +106,117 @@ renderHeaderItem.propTypes = PropTypes.oneOfType([
+			>{renderHeaderItem(headerItem)}</a>
 		);
 	});
 
 	if (caret) {
-		headerItems.push(
-			<Icon
-				key={header.length}
-				className={css.caret}
-				name={'talend-caret-down'}
-			/>,
-		);
+		const defaultCaret = (<Icon
+			key={header.length}
+			className={css.caret}
+			name={'talend-caret-down'}
+		/>);
+		headerItems.push(defaultCaret);
 	}
-
 	return headerItems;
 }
 
-/**
- * @param {object} props react props
- * @example
- const props = {
-	header: [
-		{
-			displayMode: 'status',
-			status: 'failed',
-			label: 'Failed',
-			icon: 'fa fa-close',
-			actions: [
-				{
-					label: 'cancel',
-					icon: 'fa fa-cancel',
-					onClick: action('onCancel'),
-				}
-			],
-		},
-		{
-			label: 'by Charles',
-			bsStyle: 'default',
-			tooltipPlacement: 'top',
-		},
-		{
-			displayMode: 'action',
-			label: 'edit',
-			icon: 'fa fa-edit',
-			onClick: action('onEdit'),
-			tooltipPlacement: 'right',
-			hideLabel: true,
-			link: true,
-		}
-	],
-	content: [
-		{
-			label: 'Content1',
-			description: 'Description1',
-		},
-		{
-			label: 'Content2',
-			description: 'Description2',
-		}
+function getKeyValueContent(content) {
+	return (content.map(
+		(item, index) => (
+			<div key={index} className={css.content}>
+				<div className={css.label}><Label>{item.label}</Label></div>
+				<div className={css.description}>{item.description}</div>
+			</div>)
+	));
+}
 
-	]
-};
- <CollapsiblePanel {...props} />
- */
-function CollapsiblePanel({ header, content }) {
-	const hasContent = content && content.length;
-	const headerItems = renderHeader({ header, caret: hasContent });
+function getTextualContent(content) {
+	return (
+		<div className={css['vertical-content']}>
+			<div className={css['content-upper']}>
+				{content.upper.map((item, index) => {
+					if (typeof item === 'string') {
+						return (<span key={index}>{item}</span>);
+					}
+					const { label, tooltipPlacement } = item;
+					return (
+						<TooltipTrigger
+							key={index}
+							label={label}
+							tooltipPlacement={tooltipPlacement}
+						>
+							<span>{label}</span>
+						</TooltipTrigger>
+					);
+				})}
+			</div>
+			<div className={css['content-text']}>
+				{content.description}
+			</div>
+		</div>
+	);
+}
 
-	if (hasContent) {
-		return (
-			<Panel className={css['tc-collapsible-panel']} collapsible header={headerItems}>
-				{content.map(
-					(item, index) => ((
-						<div key={index} className={css.content}>
-							<div className={css.label}><Label>{item.label}</Label></div>
-							<div className={css.description}>{item.description}</div>
-						</div>
-						)),
-					)}
-			</Panel>
-		);
+function CollapsiblePanel({ header, content, onSelect, selected }) {
+	const headerItems = renderHeader({ header, caret: content, onSelect });
+	const className = classNames(css['collapsible-panel'],
+		onSelect && css.selectable,
+		selected && css['selected-panel']);
+	const noContentClassName = classNames(
```

Comment:
`className` and `noContentClassName` has things in common and other are depending on the fact that there is a content or not.
This can be merged, using `classNames` lib
```
const componentClassNames = {
    myClassName: condition,
    myOtherClassName: myOtherCondition,
    myCommonClassName,
}
```